cluster/addons/hp-client/hugepages/hpIsolator.go

```diff
@@ -0,0 +1,163 @@
+package hugepages
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"strconv"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/cluster/addons/hp-client/hugepages/discovery"
+	"k8s.io/kubernetes/cluster/addons/iso-client/opaque"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/api/v1/helper"
+	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/lifecycle"
+)
+
+type hugepageIsolator struct {
+	name			string
+	hpTotal			int
+}
+
+type isoSpec struct {
+	Hugepage string `json:"hugepage"`
+}
+
+const (
+	MegaBytesToBytes = 1048576
+)
+
+// implementation of Init() method in  "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/lifecycle".Isolator
+func (h *hugepageIsolator) Init() error {
+	return opaque.AdvertiseOpaqueResource(h.name, h.hpTotal)
+}
+
+// Constructor for Isolator
+func New(name string) (*hugepageIsolator, error) {
+	totalHPAvail, err := discovery.DiscoverHugepageNumber()
+	return &hugepageIsolator{
+		name:		name,
+		hpTotal:	totalHPAvail,
+	}, err
+}
+
+func (h *hugepageIsolator) gatherContainerRequest(container v1.Container) int64 {
+	resource, ok := container.Resources.Requests[helper.OpaqueIntResourceName(h.name)]
+	if !ok {
+		return 0
+	}
+	return resource.Value()
+}
+
+func (h *hugepageIsolator) countHugepagesFromOIR(pod *v1.Pod) int64 {
+        var hpAccu int64
+        for _, container := range pod.Spec.Containers {
+                hpAccu = hpAccu + h.gatherContainerRequest(container)
+        }
+        return hpAccu
+}
+
+func getHugepageSize(pod *v1.Pod) (string, error) {
+	glog.Infof("Inside getHugepageSize function Value of pod.Annotations is %v", pod.Annotations)
+	if pod.Annotations["pod.alpha.kubernetes.io/isolation-api"] != "" {
+		spec := &isoSpec{}
+		err := json.Unmarshal([]byte(pod.Annotations["pod.alpha.kubernetes.io/isolation-api"]), spec)
+		if err != nil {
+			glog.Fatalf("Cannot unmarshal isoSpec: %v", err)
+			return "", err
+		}
+		if spec.Hugepage != "" {
+			return spec.Hugepage, nil
+		}
```

Comment:
you can move annotations parsing to another function, so each of your function will be responsible for one task :)